lib/sqlite3/database.rb

```diff
@@ -691,6 +691,19 @@ def readonly?
       @readonly
     end
 
+    def busy_handler_timeout=( milliseconds )
+      timeout_seconds = milliseconds.fdiv(1000)
+      retry_interval = 0.001 # 1 millisecond
+
+      busy_handler do |count|
+        @start_time = Process.clock_gettime(Process::CLOCK_MONOTONIC) if count == 0
+
+        next sleep(retry_interval) unless (count % 100) == 0
```

Comment:
I'm not even sure this module thing is worth it. `Process.clock_gettime(Process::CLOCK_MONOTONIC)` is really really fast, it's not even really a syscall as it uses VDSO so never leaves user space (on Linux).

I just benchmarked it, it's off course slower than doing a modulo, but it's in the same order of magnitude: https://gist.github.com/casperisfine/72af4812a0cc8c925432a5d2722bc12a

So given that every sleep could be very long, I don't think this potential for innacuracy is worth this micro-optimization. You can just pre-compute the deadline like in my benchmark, and that should be more than fast enough.

Also if we're going for micro-optimizations, might as well define that instance variable early to ensure a consistent shape.

Reply:
@byroot: Makes sense. The win from releasing the GVL is the core detail, and any other micro-optimizations should be considered later based on real-world usage. I simplified the handler in [d3dcd52](https://github.com/sparklemotion/sqlite3-ruby/pull/443/commits/d3dcd52c753462653d2584f93fea0db43018d172)